packages/video_player/video_player/lib/video_player.dart

```diff
@@ -340,19 +356,18 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
   }
 
   @override
+  // ignore: must_call_super
   Future<void> dispose() async {
     if (_creatingCompleter != null) {
-      await _creatingCompleter.future;
+      await _creatingCompleter!.future;
       if (!_isDisposed) {
         _isDisposed = true;
         _timer?.cancel();
         await _eventSubscription?.cancel();
         await _videoPlayerPlatform.dispose(_textureId);
       }
-      _lifeCycleObserver.dispose();
     }
-    _isDisposed = true;
-    super.dispose();
+    _lifeCycleObserver.dispose();
```

Comment:
Looks like the logic here is changed, is it related to null-safety migration?

Reply:
Fixed.